crates/oxc_linter/src/rules/typescript/no_this_alias.rs

```diff
@@ -94,7 +94,7 @@ impl Rule for NoThisAlias {
                 }
 
                 if let BindingPatternKind::BindingIdentifier(identifier) = &decl.id.kind {
-                    if !self.allow_names.contains(&identifier.name.to_string()) {
+                    if !self.allow_names.contains(&identifier.name) {
```

Comment:
I think `as_str` should work here. Strings are really hard to work with isn't it 😅

Reply:
I just made the config become an atom which should be the cleanest?